www/perl-lib/SMT/Rest/SCCAPIv4.pm

```diff
@@ -335,6 +335,24 @@ sub update_product
 
     my $regs = SMT::Utils::lookupRegistrationByGUID($self->dbh(), $guid, $self->request());
 
+    if (SMT::Utils::hasRegSharing()) {
+        my $product = SMT::Utils::lookupProductById($self->dbh(), $req_pdid);
+
+        if ($product->{product_type} eq "base") {
+            # disallow base product change
+            return (
+                Apache2::Const::HTTP_UNPROCESSABLE_ENTITY,
+                "No installed product with requested migration target product found"
+            ) unless (SMT::Utils::checkMigrationPath($self->dbh(), $req_pdid, $regs));
+        } else {
+            # require extension dependency to be activated
+            return (
+                Apache2::Const::HTTP_UNPROCESSABLE_ENTITY,
+                "Product dependency activation not found"
```

Comment:
Could this error be improved?

Something like "No activation for dependency of product $xyz"?

Reply:
That doesn't mean it should be incomprehensible when you do make that mistake.

Reply:
Otherwise, lgtm